geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionStats.java

```diff
@@ -1188,11 +1210,51 @@ public void endPutLocal(long start) {
     stats.incLong(putLocalTimeId, delta);
   }
 
+  public long startBucketClear() {
+    stats.incLong(bucketClearsInProgressId, 1);
+    return clock.getTime();
+  }
+
+  public void endBucketClear(long start) {
+    long delta = clock.getTime() - start;
+    stats.incLong(bucketClearsId, 1);
+    stats.incLong(bucketClearsInProgressId, -1);
+    stats.incLong(bucketClearTimeId, delta);
+  }
+
   public void incPRMetaDataSentCount() {
     this.stats.incLong(prMetaDataSentCountId, 1);
   }
 
   public long getPRMetaDataSentCount() {
     return this.stats.getLong(prMetaDataSentCountId);
   }
+
+  public long getBucketClearCount() {
+    return stats.getLong(bucketClearsId);
+  }
+
+  public void incBucketClearCount() {
```

Comment:
This code is only used in test. why?

Reply:
increase bucket clear count is actually done in endBucketClear(). Your incBucketClearCount() should be removed, or you can change your endBucketClear() to call incBucketClearCount().

Reply:
It is used now.

Reply:
It looks like it is only used by tests. Shouldn't this method be removed to force callers to use startBucketClear and endBucketClear?

Reply:
It was done that way in working with Anil to enable testing. I don't think it is necessary. I will pull it.

Reply:
done